pkg/kv/kvserver/kvstorage/storage.go

```diff
@@ -121,8 +126,18 @@ type Engines struct {
 // MakeEngines creates an Engines handle in which both state machine and log
 // engine reside in the same physical engine.
 func MakeEngines(eng storage.Engine) Engines {
-	// TODO(#158281): in test builds, wrap the engines in a way that allows
-	// verifying that all accesses touch the correct subset of keys.
+	if util.RaceEnabled {
+		// Wrap the engines with span set engines to catch incorrect engine
+		// accesses.
+		wrappedStateEngine := spanset.NewSpanSetEngine(eng, validateIsStateEngineSpan)
+		wrappedLogEngine := spanset.NewSpanSetEngine(eng, validateIsRaftEngineSpan)
```

Comment:
Final touch / brevity nits:

`spanset.NewEngine` (typical Go thing is to remove parts of the name that are already in the package name)

Also don't need the vars:

```golang
return Engines {
	stateEngine: spanset.NewEngine(eng, validateIsStateEngineSpan),
	logEngine:   spanset.NewEngine(eng, validateIsRaftEngineSpan),
	todoEngine: eng.
}
```

Same below.